stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2454,26 +2449,57 @@ internal struct _${Self}StorageHeader<${TypeParameters}> {
 
   // Placeholder values until the struct can be initialized.
   internal var initializedEntries: _UnsafeBitMap! = nil
-  internal var keys: UnsafeMutablePointer<Key>! = nil
+  internal var keys: UnsafeMutableRawPointer! = nil
 % if Self == 'Dictionary':
-  internal var values: UnsafeMutablePointer<Value>! = nil
+  internal var values: UnsafeMutableRawPointer! = nil
 % end
 }
 
-/// An instance of this class has all `${Self}` data tail-allocated.
-/// Enough bytes are allocated to hold the bitmap for marking valid entries,
-/// keys, and values. The data layout starts with the bitmap, followed by the
-/// keys, followed by the values.
-@_versioned
-final internal class _Native${Self}Storage<${TypeParameters}> :
-  ManagedBuffer<_${Self}StorageHeader<${TypeParameters}>, UInt8> {
+internal class _RawNative${Self}Buffer : 
+  ManagedBuffer<_${Self}StorageHeader, UInt8> {
+  internal typealias StorageHeader = _${Self}StorageHeader
+  internal typealias RawBuffer = _RawNative${Self}Buffer
+
+  internal var _body: StorageHeader
+
+  // This API is unsafe and needs a `_fixLifetime` in the caller.
```

Comment:
We generally try to expose a `withXXX` method to make `_fixLifetime`s less necessary.